assets/js/leaves.js

```diff
@@ -46,38 +49,84 @@ class Leaf {
   static selectedClass = "select";
   static highlightClass = "highlight";
 
-  static bindHandlers() {
-    // bind a delegated click handler to override tag link behavior
-    const asideContainer = document.querySelector("aside");
-    asideContainer.addEventListener("click", (event) => {
+  constructor(panel, ignore_ids) {
+    // store a reference to the panel object
+    this.panel = panel;
+    this.container = document.querySelector("aside");
+    this.bindHandlers();
+    // list of ids to ignore when loading leaf details
+    // (known non-leaf elements, including branch start targets)
+    this.ignore_ids = ignore_ids || [];
+  }
+
+  static isTag(element) {
+    // check if an element is a tag
+    return (
+      element.tagName == "A" && element.parentElement.classList.contains("tags")
+    );
+  }
+
+  bindHandlers() {
+    // bind a delegated click handler to override tag link behavior;
+    // delegated so it applies to tags in leaf details loaded after bound
+    this.container.addEventListener("click", (event) => {
       let element = event.target;
       // if click target is a link in the tags section, select leaves for that tag
-      if (
-        element.tagName == "A" &&
-        element.parentElement.classList.contains("tags")
-      ) {
+      if (Leaf.isTag(element)) {
         event.preventDefault();
         event.stopPropagation();
-        Leaf.setCurrentTag(element.dataset.tag);
+        this.currentTag = element.dataset.tag;
         element.classList.add(Leaf.selectedClass);
-        asideContainer.dispatchEvent(TagSelectEvent);
+        this.container.dispatchEvent(TagSelectEvent);
       }
     });
 
     // bind handler to current tag x button to deactivate tag
     const activeTagClose = document.querySelector("#current-tag .close");
-    activeTagClose.addEventListener("click", (event) => {
-      Leaf.setCurrentTag();
-      asideContainer.dispatchEvent(TagDeselectEvent);
+    if (activeTagClose) {
+      activeTagClose.addEventListener("click", (event) => {
+        this.currentTag = null;
+        this.container.dispatchEvent(TagDeselectEvent);
```

Comment:
Does this mean that in the end we needed to create specific listeners for the tag functionality to work? Or are these listeners just needed for the "X" buttons, etc?

Reply:
We do have a custom click handler for the tags since they have a specific behavior in the context of the tree (filter active leaves, display active tag). This handler is for closing the active tag, though.